### DISCUSSION TITLE
refresh_token is revoked

### DISCUSSION THREAD

**Discussion by stalno** [etsy/open-api#922] [Q&A]
Hello.
I made my own app that updates listings stocks.
Its update every 3 hours from our wms system.
But after 2-3 days of working i received a message, that my refresh token is revoked.
Password nobody change, everything ok.
What the problem it can be?

**Comment by subash047** [comment]
Hello, 
We are facing a similar issue. Some of our clients are facing the same issue of the refresh token is revoked.

@nateswart @capndesign  
Can we have more details, on what cases Etsy store owners face the above issue while connected using a third-party Etsy app?

**Comment by Gareth064** [comment]
etsychristensen doesn't work for Etsy anymore, he left a couple of months ago.

If you want help from Etsy Devs then you need to email them. They don't check this github much anymore.


Kind Regards

Gareth Doherty



It’s nice to be important,

But it’s more important to be nice.



________________________________

**Reply by stalno** [reply]
I got it. I will.

**Reply by PMStanley** [reply]
@stalno If you get a response would you please share it here so others can reference this post if it ever happens to them

**Comment by emergingdzns** [comment]
When you fetch the ACCESS token it gives a new refresh token every time. Are you updating your local records with that new refresh token? Open Refresh Tokens expire after 90 days. So if you aren't updating your database with the latest refresh token, you're probably hitting the 90 day limit.

Just a guess.

**Reply by stalno** [reply]
No.
Before every stock update ( every 3 hours ), i request ACCESS token, store in db new refresh token, and than update stocks. Algorithm is rather clear.

**Comment by dave-shawley** [comment]
I've been facing a similar issue (see [related discussion](https://github.com/etsy/open-api/discussions/744)).  It seems that if the response body contains some breadcrumbs.  The variations that I have seen thus far are:

| error | error_description | Suspected cause | Remediation |
| --- | --- | --- | --- |
| invalid_grant | refresh_token is revoked | User changed their Etsy password or otherwise revoked access to the application | Re-authorize application |
| invalid_token | access token is expired | Simple access token expiration | Refresh the token using the last `refresh_token` |
| invalid_token | access token is invalid | The access token is not valid | Try to refresh the token ... _I never got to the bottom of this case_ |

There was a [GitHub issue](https://github.com/etsy/open-api/issues/519#issuecomment-1168806286) that had more breadcrumbs but it was deleted when they migrated to discussions 😞 

My implementation currently stores the access & refresh tokens on each refresh.  Once I receive the `refresh_token is revoked` error, the OAuth 2 connection is no longer valid and needs to be re-established by the Etsy customer.  Previous discussions about the `refresh_token is revoked` error were always connected to the Etsy customers password being changed or explicitly removing the applications connection to the store.

HTH
